hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java

```diff
@@ -141,6 +145,9 @@ public class MirrorNodeEvmProperties implements EvmProperties {
     @DurationMin(seconds = 100)
     private Duration rateLimit = Duration.ofSeconds(100L);
 
+    @Getter
+    private TreeMap<String, Long> evmVersions = new TreeMap<>();
```

Comment:
Invert the key and value then you can use `evmVersions.floorEntry(blockNum)` to efficiently find the value instead of looping.

```suggestion
    private NavigableMap<Long, String> evmVersions = new TreeMap<>();
```

Reply:
Comment addressed. I've inverted the key and value for better search.